woocommerce/woocommerce
# Validate missing fields in Store API

### Changes proposed in this Pull Request:

In additional fields, we would only validate passed fields, this means required missing fields would not get validated, this fixes it and enable the test for it.

Closes https://github.com/woocommerce/woocommerce/issues/45496



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Tests should pass.
2. Add the following snippet to your store
```php
add_action(
	'woocommerce_loaded',
	function () {
		__experimental_woocommerce_blocks_register_checkout_field(
			array(
				'id'       => 'plugin-namespace/job-function',
				'label'    => 'What is your main role at your company?',
				'location' => 'additional',
				'required' => true,
				'type'     => 'text',
			)
		);
	}
);
```
3. Send this request to Store API after adding to cart:
```json
{
	"billing_address": {
		"first_name": "test",
		"last_name": "test",
		"company": "",
		"address_1": "test",
		"address_2": "",
		"city": "test",
		"state": "",
		"postcode": "cb241ab",
		"country": "GB",
		"phone": "",
		"email": "testaccount@test.com"
	},
	"shipping_address": {
		"first_name": "test",
		"last_name": "test",
		"company": "",
		"address_1": "test",
		"address_2": "",
		"city": "test",
		"state": "",
		"postcode": "cb241ab",
		"country": "GB",
		"phone": ""
	},
	"payment_method": "bacs",
	"additional_fields": {}
}
```
4. You should get an error saying the field is missing or not the correct type.
